src/cplscheme/CouplingData.hpp

```diff
@@ -25,44 +46,52 @@ struct CouplingData { // @todo: should be a class from a design standpoint. See
     return data->values();
   }
 
-  /// Data values of previous iteration (1st col) and previous time windows.
-  DataMatrix oldValues;
+  void storeIteration()
+  {
+    _lastIteration = this->values();
+  }
 
-  mesh::PtrData data;
+  const Eigen::VectorXd lastIteration() const
+  {
+    return _lastIteration;
+  }
 
-  mesh::PtrMesh mesh;
+  [[deprecated("lastIteration should be read only!")]] void writeLastIteration(Eigen::VectorXd value)
+  {
+    _lastIteration = value;
+  }
```

Comment:
That's not meant to be merged, right?

Reply:
Yes. Removed it in dd75e064837846d3da99deed1f8ed12ea6e841db.